docs/porting.md

```diff
@@ -211,7 +203,41 @@ This wouldn't help a case where each thread having a copy of the cache would be
 prohibitive, but it does fix possible issues with resource leaks issues due to
 races filling a cache.
 
-### Making mutable global caches thread-safe with locking
+### Read-copy-update
+
+[Read-Copy-Update (RCU)](https://en.wikipedia.org/wiki/Read-copy-update) is a thread safe pattern to implement lock-free data structures. It is useful when reads are much more frequent than writes. It is commonly used for caching, where reads are frequent and writes are infrequent.
+
+Consider a library which generates n'th Fibonacci number. The library caches previously computed Fibonacci numbers.
+
+```python
+cache = [0, 1]
+
+
+def fib(nth: int) -> int:
+    global cache
+    if nth < 1:
+        raise ValueError("nth must be a positive integer")
+    # Read from the existing cache
+    if nth <= len(cache):
+        return cache[nth - 1]
+
+    # Create a new copy of the cache
+    new_cache = cache.copy()  # list.copy is atomic
+    for i in range(len(new_cache), nth + 1):
+        new_cache.append(new_cache[i - 1] + new_cache[i - 2])
+
+    # Update the reference to point to the new cache
+    cache = new_cache
+    return cache[nth - 1]
```

Comment:
This should `return new_cache[nth - 1]`. If we are doing `fib(10)` and another thread is doing `fib(9)` then the other thread might shorten `cache` in between these two lines.